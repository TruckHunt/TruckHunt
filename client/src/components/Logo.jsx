import React from 'react';
import { Navbar } from 'react-bootstrap';
import { Link } from 'react-router-dom';

const Logo = () => (
  <Navbar.Brand>
    <Link to="/" >
      <div className="logo-inline">
        <img className="logo-img" src={'https://s3-us-west-1.amazonaws.com/zollstorage/thesis/LogoV1.png'} alt="logo" />
        <h2 className="logo-text">  Truck Hunt</h2>
      </div>
    </Link>
  </Navbar.Brand>
);

export default Logo;
