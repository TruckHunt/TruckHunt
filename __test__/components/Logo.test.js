import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import React from 'react';
import { StaticRouter } from 'react-router-dom';
import Logo from '../../client/src/components/Logo.jsx';


describe('Logo Component Tests', () => {
  let logo;
  beforeEach(() => {
    logo = shallow(<Logo />);
  });

  it('should render without crashing', () => {
    expect(logo.exists()).toBe(true);
  });

  it('should render an image"', () => {
    expect(logo.find('img').length).toEqual(1);
  });

  test('Logo matches snapshot', () => {
    const component = renderer.create(
      <StaticRouter>
        <Logo />
      </StaticRouter>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
