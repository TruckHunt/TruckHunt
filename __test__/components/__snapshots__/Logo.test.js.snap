// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo Component Tests Logo matches snapshot 1`] = `
<a
  className="navbar-brand"
  href="/"
  onClick={[Function]}
>
  <div
    className="logo-inline"
  >
    <img
      alt="logo"
      className="logo-img"
      src="https://s3-us-west-1.amazonaws.com/zollstorage/truckhunt/LogoV2.png"
    />
  </div>
</a>
`;
